lib/angular/shared/geolocation/geolocation-spec.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+/* globals inject */
+/*global navigator:true*/
+
+var sinon = require('sinon'),
+  angular = require('angular'),
+  //should = require('should'),
+  initModule = require('../../../../test/initModule');
+require('angular-mocks');
+
+describe("Test Geolocation Service", function() {
+  //Setting up a mock module to test.
+  before(function() {
+    initModule();
+    // Import the geolocation service
+    require('./geolocationService');
+  });
+
+  beforeEach(angular.mock.module('wfm.appform'));
+  beforeEach(inject(function(GeolocationService) {
+    this.GeolocationService = GeolocationService;
+  }));
+
+  it('The getGeolocation service method should return coordinates when navigator is available and an error message when not available', function() {
+
+    // setup stub
+    var sinonStub = sinon.stub();
+    navigator.geolocation = {};
+    navigator.geolocation.getCurrentPosition = sinonStub;
+
+    // mock success/error responses
+    var expectedResponse = {coords: {latitude: 52.251198099999996, longitude : -7.1534572999999995}};
+    var mockErr = {code: 1, message: "User denied Geolocation"};
+
+    // Stubs the method only for the provided arguments which must be functions
+    var stubMethod = sinonStub.withArgs(sinon.match.func, sinon.match.func);
+
+    // specify arguments to pass to the callback on call n
+    stubMethod.onCall(0).callsArgWith(0, expectedResponse);
```

Comment:
Can you move the setup logic for the stub to where the stub is defined. It's easier to read.